core/server/master/src/main/java/alluxio/master/lineage/LineageMasterFactory.java

```diff
@@ -50,10 +50,10 @@ public String getName() {
   @Override
   public LineageMaster create(MasterRegistry registry, JournalFactory journalFactory) {
     Preconditions.checkArgument(journalFactory != null, "journal factory may not be null");
-    LOG.info("Creating {} ", LineageMaster.class.getName());
+    LOG.info("Creating {} ", DefaultLineageMaster.class.getName());
     FileSystemMaster fileSystemMaster = registry.get(FileSystemMaster.class);
-    LineageMaster lineageMaster = new LineageMaster(fileSystemMaster, journalFactory);
-    registry.add(LineageMaster.class, lineageMaster);
+    LineageMaster lineageMaster = new DefaultLineageMaster(fileSystemMaster, journalFactory);
+    registry.add(DefaultLineageMaster.class, lineageMaster);
```

Comment:
`registry.add(LineageMaster.class, lineageMaster);`
Here we register implementations based on its type (interface)